### DISCUSSION TITLE
Many to Many field with a context field which is another relation?

### DISCUSSION THREAD

**Discussion by wiz256** [directus/directus#22356] [Feature Requests]
### Summary

I have collections:

`cities` (id, name)
`routes` (id, name)
`routeCityStops` (id, routeId, cityId, address, departureTime)
`routeFares` (routeId, cityFromId, cityToId, priceValue)

What is the best way to show in `routes` item details all `routeFares` belonging to this route as a table with ability to add/modify routeFare records?

Many2Many did not help me because `routeFares` has 3 relations (because of cityFromId, cityToId and routeId).

How to properly configure such setup for model with Route and RouteFares?

### Motivation

Need to support "tripple" relationship like for `routeCityStops` in:
`cities` (id, name)
`routes` (id, name)
`routeCityStops` (id, routeId, cityId, address, departureTime)
`routeFares` (routeId, cityFromId, cityToId, priceValue)

### Requirements List

Basically I want:
1. being on `route` item details page to see all `routeFares` related to this route with cityFrom, cityTo and price values
2. when new `routeFares` record is added from `routes` item details page then `routeFares` should get `routeId` automatically set (as we are on that route details page already).
3. when editing or creating new `routeFares` record from either from within `routes` item details page of from `routeFares` list page we should be able to select only those fromCityId and toCityId which were assigned as available stops through `routeCityStops` collection.

### Alternatives

Maybe there is already some easy way how to configure it.
